src/main/java/ua.horobets/hw19/JDBC.java

```diff
@@ -0,0 +1,104 @@
+package ua.horobets.hw19;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class JDBC implements Dao<Person> {
+    private static final JDBC INSTANCE = new JDBC();
+
+    private JDBC() {
+    }
+
+    public static JDBC getInstance() {
+        return INSTANCE;
+    }
+
+    private static final String SAVE_SQL = "INSERT INTO persons(first_name,last_name,email) VALUES (?,?,?);";
+    private static final String FIND_SQL = "SELECT * FROM persons WHERE id= ?;";
+    private static final String FIND_ALL = "SELECT * FROM persons ORDER BY id;";
+    private static final String UPDATE = "UPDATE persons SET first_name=?,last_name=?,email=? WHERE id=?;";
+    private static final String DELETE = "DELETE FROM persons WHERE id=?;";
+
+    @Override
+    public Optional<Person> findByID(long id) {
+        try (Connection connection = JDBCConnection.get()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(FIND_SQL);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                Person person = new Person(
+                        resultSet.getLong("id"),
+                        resultSet.getString("first_name"),
+                        resultSet.getString("last_name"),
+                        resultSet.getString("email")
+                );
```

Comment:
In findAll method you duplicate this part of the code.
You can move it to separate method